greenriver/hmis-frontend
# 6064-sentry-grouping

## Description

[GH issue](https://github.com/open-path/Green-River/issues/6064)

* upgrade sentry 7->8, add reactRouter integration
* add apollo-link-sentry to improve sentry instrumentation for graphql errors 
* adjust apollo error link handling to improve grouping and reporting
* upgrade @sentry/vite-plugin, configure to include component names
* code organization: pull sentry config out of index.tsx, extract errorLink from apollo link

### Testing
for testing, I set `PUBLIC_SENTRY_DSN` to the development sentry project

### Testing case auth failure
1) login
1) log your user out on the server from the console `Hmis::User.where(id:1).update_all(unique_session_id:'')`
1) on api request: error should not be reported to sentry, user prompted to sign in
### Testing server error
1) login and visit /projects
1) adjust graphqlController#execute to return `return render status: 500, json: { errors: [ { message: 'test' }, ], data: {}, }`
1) click on a project
1) 1 "graphql error" should be logged to sentry. Message should contain informative breadcrumbs




## Type of change
[//]: # 'remove options that are not relevant'

## Checklist before requesting review
